### PR TITLE
feat: separated db encryption into a separate plugin

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

*  **BREAKING_CHANGE:** `DBEncryption` from `rudder_plugin_interface` has been replaced with `RudderDBEncryption` from the `rudder_plugin_db_encryption`
* Separated the DB Encryption feature into a separate plugin.
* Added a new DBEncryptionInterface in the rudder_plugin_interface which was actually implemented by the rudder_plugin_db_encryption.
* Made iOS Modules of Appcenter, Appsflyer, Firebase Flutter Device Modes as `static_frameworks` to make them support for linking either dynamically and statically.
* Zetetic Dependencies on the Android platform need not be added manually now as they come by default with the plugin.
* RudderDatabaseEncryption pod need not be added manually now as it comes default with the plugin.
* Added Proguard rules to the Sample App to support Proguard Full Mode.
### The DBEncryption feature can be enabled by following the below code snippet: 
```dart
RudderDBEncyrption dbEncryption = RudderDBEncyrption(true, "password");
    MobileConfig mc = MobileConfig(
      dbEncryption: dbEncryption,
    );

RudderConfigBuilder builder = RudderConfigBuilder();
builder
    ..withDataPlaneUrl("DATA_PLANE_URL")
    ..withMobileConfig(mc)
    ..withLogLevel(RudderLogger.VERBOSE);
rudderClient.initialize("WRITE_KEY",  config: builder.build(), options: null);
```

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [x] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: packages/features/rudder_plugin_db_encryption/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## 0.0.1
+
+* TODO: Describe initial release.
```
Comment (bardisg) [inline packages/features/rudder_plugin_db_encryption/CHANGELOG.md:3]: please complete this TODO

File: packages/features/rudder_plugin_db_encryption/LICENSE [hunk 2]
```diff
@@ -0,0 +1 @@
+TODO: Add your license here.
```
Comment (bardisg) [inline packages/features/rudder_plugin_db_encryption/LICENSE:1]: please complete this TODO

File: packages/features/rudder_plugin_db_encryption/ios/rudder_plugin_db_encryption.podspec [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+Pod::Spec.new do |s|
+  s.name             = 'rudder_plugin_db_encryption'
+  s.version          = '0.0.1'
+  s.summary          = 'A new Flutter plugin project.'
+  s.description      = <<-DESC
+A new Flutter plugin project.
+                       DESC
+  s.homepage         = 'http://example.com'
+  s.license          = { :file => '../LICENSE' }
+  s.author           = { 'Your Company' => 'email@example.com' }
```
Comment (bardisg) [inline packages/features/rudder_plugin_db_encryption/ios/rudder_plugin_db_encryption.podspec:14]: can we add real values here?

File: packages/features/rudder_plugin_db_encryption/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  flutter_lints: ^2.0.0
+
+flutter:
+  plugin:
+    platforms:
+      android:
+        package: com.example.rudder_plugin_db_encryption
```
Comment (bardisg) [inline packages/features/rudder_plugin_db_encryption/pubspec.yaml:30]: can we add relal values here?

**Review by bardisg — COMMENTED** [review]
How plugin and features package are different?

File: packages/features/rudder_plugin_db_encryption/LICENSE (hunk 2 above)
Comment (desusai7) [reply packages/features/rudder_plugin_db_encryption/LICENSE]: Completed the TODO.

File: packages/features/rudder_plugin_db_encryption/ios/Classes/RudderPluginDbEncryptionPlugin.m [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    if ([@"addDBEncryptionToConfig" isEqualToString:call.method]) {
+        NSDictionary* dbEncryptionDict = [call.arguments objectForKey:@"dbEncryption"];
+        if(dbEncryptionDict == nil)
+            return;
+        
+        BOOL enabled = [[dbEncryptionDict objectForKey:@"enabled"] boolValue];
+        NSString* encryptionKey = [dbEncryptionDict objectForKey:@"key"];
+        
+        if (encryptionKey == nil || [encryptionKey length] == 0)
+            return;
```
Comment (pallabmaiti) [inline packages/features/rudder_plugin_db_encryption/ios/Classes/RudderPluginDbEncryptionPlugin.m:24]: Can we add a log here?

File: packages/features/rudder_plugin_db_encryption/ios/rudder_plugin_db_encryption.podspec [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+Pod::Spec.new do |s|
+  s.name             = 'rudder_plugin_db_encryption'
+  s.version          = '1.0.0'
+  s.summary          = 'iOS Module of the RudderStack SDK for Flutter, to enable database encryption.'
+  s.description      = <<-DESC
+                        This module enables database encryption for the RudderStack SDK for Flutter.
+                       DESC
+  s.homepage         = 'https://github.com/rudderlabs/rudder-sdk-flutter'
+  s.license          = { :file => '../LICENSE' }
+  s.author           = { 'RudderStack' => 'sdk@rudderstack.com' }
```
Comment (pallabmaiti) [inline packages/features/rudder_plugin_db_encryption/ios/rudder_plugin_db_encryption.podspec:14]:
```suggestion
  s.author           = { 'RudderStack' => 'sdk-accounts@rudderstack.com' }
```

**Review by pallabmaiti — CHANGES_REQUESTED** [review]

File: packages/features/rudder_plugin_db_encryption/ios/Classes/RudderPluginDbEncryptionPlugin.m (hunk 5 above)
Comment (desusai7) [reply packages/features/rudder_plugin_db_encryption/ios/Classes/RudderPluginDbEncryptionPlugin.m:24]: Sure, will add that

Comment (desusai7) [reply packages/features/rudder_plugin_db_encryption/ios/Classes/RudderPluginDbEncryptionPlugin.m]: Added log for the same

File: packages/features/rudder_plugin_db_encryption/ios/rudder_plugin_db_encryption.podspec (hunk 6 above)
Comment (desusai7) [reply packages/features/rudder_plugin_db_encryption/ios/rudder_plugin_db_encryption.podspec]: Changed the email address across all the podspec files.

**Review by bardisg — APPROVED** [review]

**Review by pallabmaiti — APPROVED** [review]

**Pull request merged**